roofline.cpp

```diff
@@ -177,14 +179,24 @@ int main(int argc, char **argv)
     int numGpuDevices;
     HIP_ASSERT(hipGetDeviceCount(&numGpuDevices));
 
-    using archs_t = std::unordered_set<std::string>;
-    archs_t supported_archs{"gfx908", "gfx90a", "gfx940", "gfx941", "gfx942"};
+    using datatypes = std::unordered_set<std::string>;
+    using archs_t = std::map<std::string, datatypes>;
+    datatypes unsupported_datatypes;
+
+    /* supported_archs_unsupported_dt indicates supported archs and the corrseponding datatypes which ARE NOT supported by each arch */
+    archs_t supported_archs_unsupported_dt = {
```

Comment:
I got your point. Just the name is still too...:)
Any way, not a big deal for now.